### PR TITLE
[percep] Merge code into main

### PR DESCRIPTION
**Pull request by mmartin4972**
* Refactor for ZED SDK 3.0
* Implemented Point Cloud Obstacle Detection
* Cleaned and Restructured Code
* Added read from and write to file capabilities for Point Cloud, Depth, and RGB data

### CODE REVIEW

**Comment by mmartin4972** [comment]
Why does it say merge branch 'main' into main_merge2? I'm pretty sure I specified it to be the other way around

File: jetson/percep/pcl.hpp
Comment (raytitan) [inline jetson/percep/pcl.hpp:42]: No labels for this print?

File: jetson/percep/perception.hpp
Comment (raytitan) [inline jetson/percep/perception.hpp:62]: What do you think of moving these into config files like nav and localization have

**Review by raytitan — CHANGES_REQUESTED** [review]
Do you need to commit the test images into the repository? We should probably only have code in here

File: jetson/percep/perception.hpp
Comment (mmartin4972) [reply jetson/percep/perception.hpp:62]: We are currently working on that

File: jetson/percep/pcl.hpp
Comment (mmartin4972) [reply jetson/percep/pcl.hpp]: Good catch

**Comment by mmartin4972** [comment]
> Do you need to commit the test images into the repository? We should probably only have code in here

I don't think we have any images in our repository? Maybe you are seeing deleted ones?

**Comment by mmartin4972** [comment]
Once I get everyone's approval I'll merge all commits after "[percep] Improved ground plane detection" into that commit

**Review by DavidSmith166 — APPROVED** [review]

**Review by amverni — DISMISSED** [review]
LGTM as long as you squash like you mentioned!

**Review by raytitan — APPROVED** [review]

**Review by DavidSmith166 — APPROVED** [review]

**Review by amverni — APPROVED** [review]

**Review by raytitan — APPROVED** [review]

**Pull request merged**
